pkg/images/uncompress.go

```diff
@@ -1,153 +0,0 @@
-package images
-
-import (
-	"fmt"
-	"os"
-	"time"
-
-	"github.com/google/go-containerregistry/pkg/authn"
-	"github.com/google/go-containerregistry/pkg/compression"
-	"github.com/google/go-containerregistry/pkg/crane"
-	v1 "github.com/google/go-containerregistry/pkg/v1"
-	"github.com/google/go-containerregistry/pkg/v1/empty"
-	"github.com/google/go-containerregistry/pkg/v1/mutate"
-	"github.com/google/go-containerregistry/pkg/v1/tarball"
-	"github.com/google/go-containerregistry/pkg/v1/types"
-)
-
-func max(a, b int) int {
-	if a > b {
-		return a
-	}
-	return b
-}
-
-func Uncompress(imageName string, dest string, auth authn.Authenticator) (string, error) {
-	var base v1.Image
-	var err error
-
-	fmt.Fprintln(os.Stderr, "fetching metadata for", imageName)
-
-	start := time.Now()
-	base, err = crane.Pull(imageName, crane.WithAuth(auth))
-	if err != nil {
-		return "", fmt.Errorf("pulling %w", err)
-	}
-	fmt.Fprintln(os.Stderr, "pulling took", time.Since(start))
-
-	img, err := uncompress(base)
-	if err != nil {
-		return "", err
-	}
-
-	start = time.Now()
-
-	err = crane.Push(img, dest, crane.WithAuth(auth))
-	if err != nil {
-		return "", fmt.Errorf("pushing %s: %w", dest, err)
-	}
-
-	fmt.Fprintln(os.Stderr, "pushing took", time.Since(start))
-
-	d, err := img.Digest()
-	if err != nil {
-		return "", err
-	}
-	image_id := fmt.Sprintf("%s@%s", dest, d)
-	return image_id, nil
-}
-
-func uncompress(base v1.Image) (v1.Image, error) {
-
-	// inspired by https://github.com/google/go-containerregistry/blob/v0.15.2/pkg/v1/mutate/mutate.go#L371
-	newImage := empty.Image
-
-	layers, err := base.Layers()
-	if err != nil {
-		return nil, fmt.Errorf("getting layers %w", err)
-	}
-
-	ocf, err := base.ConfigFile()
-	if err != nil {
-		return nil, fmt.Errorf("getting original config file %w", err)
-	}
-
-	addendums := make([]mutate.Addendum, max(len(ocf.History), len(layers)))
-
-	startUncompressing := time.Now()
-
-	var historyIdx, addendumIdx int
-	for layerIdx := 0; layerIdx < len(layers); addendumIdx, layerIdx = addendumIdx+1, layerIdx+1 {
-		startLayer := time.Now()
-		newLayer, err := uncompressedLayer(layers[layerIdx])
```

Comment:
should we also keep the uncompressed version, or did we decide that was strictly worse?